### ISSUE TITLE
ASP.NET Identity Cookie across subdomains

### ISSUE THREAD

**Issue by chancewen** [aspnet/Identity#442]
I am working on a project of asp.net mvc 5, that need single sign-on for different sub domain.

I have found the solution on stackoverfow:

```
app.UseCookieAuthentication(new CookieAuthenticationOptions
{
  AuthenticationType = "Application",
  LoginPath = "/Account/Login",
  CookieDomain = ".myapp.com"
}); 
```
However, it seems that only if the web sites of different sub domain use the same application pool in IIS, they can single sign on. However, if they use different application pools, it does not work.

Any one can help ?

And further more, is it possible to implement single sign-on for web sites that cross domain, not only cross SUB-domain , with ASP.NET Identity Framework?

Thank you very much!

**Comment by stephenstroud** [comment]
"And further more, is it possible to implement single sign-on for web sites that cross domain, not only cross SUB-domain , with ASP.NET Identity Framework?"
Using Machine KEYS?

**Comment by chancewen** [comment]
@stephenstroud thank you very much!  I forgot the mchine keys. It works after setting MachineKeys in web.config.

However, how about cross domain (not sub-domain) single sign-on with ASP.NET Identity  Framework? it seems must depend on some third party framework, such as Thinktecture?

**Comment by blowdart** [comment]
@chancewen Correct, we don't provide federated identity services in ASP.NET. The Identity Framework isn't designed for that.

**Comment by brockallen** [comment]
FWIW, there are open source projects that do support single sign-on:

https://github.com/IdentityServer/IdentityServer3

**Comment by stephenstroud** [comment]
@chancewen  I achieved this cross domain by having a standalone identity web api exposed again with machine keys.

**Comment by chancewen** [comment]
@stephenstroud Could you please share some your code about cross domain SSO without 3rd party framework? I'm really interested in it. Thank you!

**Comment by chancewen** [comment]
@brockallen thank you! I will try.

**Comment by stephenstroud** [comment]
@chancewen A template already exists: New Project > ASP.NET Web Application > Web API > Change Authentication(Defaulted to Individual User Accounts) - Just host and call from a separate domain.  

Using and consuming the implementation:
http://www.asp.net/web-api/overview/security/individual-accounts-in-web-api

Don't forget to use CORS. Hope this is what you are looking for? other than that you have brockallen's solution.

**Comment by chancewen** [comment]
@stephenstroud  Thank you very much!

**Comment by ursenix** [comment]
Excellent... seems I got an answer... Will workout this now

**Comment by BPA8** [comment]
@chancewen @stephenstroud  - I stumbled upon this thread while I was looking to implement SSO between my MVC applications which uses WEB API with Identity . WEB API, MVC 1 , MVC 2 apps deployed to azure web apps . Locally SSO seems to be working as expected but doesn't seem to be working in Azure? I have enabled useCors , added machine keys to all 3 of them . what am I missing? any pointers greatly appreciated!
